From 1e56bf8e6a5e004131fa35d9757199164026b7de Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Wed, 24 Jul 2013 12:15:08 +0700
Subject: Dalvik: Add null check to SendTraceInfoToVTune for coverity tool

BZ: 124408

Coverity tool reports the possible crash due to dereferencing the null pointer.
The patch adds a null check to make Coverity be happy.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I296d9a79aad57c94d1b4979def8ec48e67564103
Orig-MCG-Change-Id: Ica9c3f17b5cc6803e81eccce880638939e59ddf1
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Reviewed-on: http://android.intel.com:8080/124377
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/Frontend.cpp |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/vm/compiler/Frontend.cpp b/vm/compiler/Frontend.cpp
index 2665a18..326882d 100644
--- a/vm/compiler/Frontend.cpp
+++ b/vm/compiler/Frontend.cpp
@@ -3003,11 +3003,17 @@ static void SendTraceInfoToVTune(CompilationUnit *cUnit, JitTraceDescription *de
     DexStringCache params_string;
     dexStringCacheInit(&params_string);
     const char* params = dexProtoGetParameterDescriptors(&desc->method->prototype, &params_string);
-    int len = strlen(params) + strlen(desc->method->name) + 3 ;
+    int len = strlen(desc->method->name) + 3;
+
+    // If params is not nil we should account it in the total length
+    if (params != 0) {
+        len += strlen(params);
+    }
+
     char* name_with_params = (char*) dvmCompilerNew(len, true);
     strcpy(name_with_params, desc->method->name);
     strcat(name_with_params, "(");
-    if (params) {
+    if (params != 0) {
         strcat(name_with_params, params);
     }
     strcat(name_with_params, ")");
-- 
1.7.4.1

